### ISSUE TITLE
Metal backend: Compound Literals Warning for _SG_CLEAR

### ISSUE THREAD

**Issue by DctrNoob** [floooh/sokol#644]
When I compile the sokol_gfx.h implementation for Metal using C++, I get the following warning (with clang 13.1.6 under macOS):
```
sokol_gfx.h:15110:5: warning: compound literals are a C99-specific feature [-Wc99-extensions]
_SG_CLEAR(_sg_state_t, _sg);
```

The corresponding section in sokol_gfx.h looks like this:
```
#if defined(SOKOL_METAL)
    // this is ARC compatible
    #if defined(__cplusplus)
        #define _SG_CLEAR(type, item) { item = (type) { }; }                 // <-- Line 15110
    #else
        #define _SG_CLEAR(type, item) { item = (type) { 0 }; }
    #endif
#else
    #define _SG_CLEAR(type, item) { memset(&item, 0, sizeof(item)); }
#endif
```

I would have expected the code in this line to be like this for C++ 11 and beyond:
`#define _SG_CLEAR(type, item) { item = { }; }  `

Or am I missing something?

**Comment by floooh** [comment]
Hmm strange, to me this looks like a C warning, not a C++ warning TBH, also that extra cast should cause the warning, even if it is redundant. I'll try to setup a minimal reproducer, since I'm not getting that warning in my own projects.

Need to take care about some linker problem in libmodplug first I just noticed (after updating to the latest Xcode version).

**Comment by DctrNoob** [comment]
To me the warning reads like "hey, you are compiling C++ but you are using C99 syntax". That's why I wrote that I would have expected no cast at all.

**Comment by floooh** [comment]
Hmm alright, are you compiling with ```-pedantic``` by chance? That's the only way I can reproduce this warning:

```
bla ➤ clang++ -c sokol.mm -std=c++11 -pedantic
In file included from sokol.mm:3:
./sokol_gfx.h:15110:5: warning: compound literals are a C99-specific feature [-Wc99-extensions]
    _SG_CLEAR(_sg_state_t, _sg);
    ^
./sokol_gfx.h:15099:48: note: expanded from macro '_SG_CLEAR'
        #define _SG_CLEAR(type, item) { item = (type) { }; }
                                               ^
1 warning generated.
```

I'll check if removing the ```(type)```, but note that at least for plain C, the sokol headers deliberately do *not* use "pedantic" standard C, because the code needs to compile both in C and C++, so warnings with -pedantic are to be expected.

**Comment by DctrNoob** [comment]
Yes, I use `-pedantic`. And removing the cast also removes the warning for me. I'll keep it removed in my fork, so that solution is ok for me.

Thanks for your feedback and the great work.

**Comment by floooh** [comment]
Yep, removing the ```(type)``` fixes it. I don't quite understand why though, since this should just a a "redundant" type cast in C++. Maybe the compiler somehow parses this as a C "type hint".

**Comment by floooh** [comment]
Ok, I also fixed it in master (f306862). A similar macro was also used in sokol_app.h.
